### PR TITLE
Adapter for daily fees and daily revenue data, for assetchain blockchain

### PR DESCRIPTION
**Pull request by theiceeman**

### CODE REVIEW

**Comment by llamatester** [comment]
The assetchain.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees assetchain.ts
    
    🦙 Running ASSETCHAIN.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Mon, 21 Oct 2024 02:33:20 GMT
    End Date:	Tue, 22 Oct 2024 02:33:20 GMT
    ---------------------------------------------------
    
    ASSETCHAIN 👇
    Backfill start time: 29/8/2020
    NO METHODOLOGY SPECIFIED
    Daily fees: 0
    Daily revenue: 0
    End timestamp: 1729564399 (2024-10-22T02:33:19.000Z)

**Comment by theiceeman** [comment]
Hello @0xngmi 
can you review this pr pls
Or is there something still missing on our end.
Thanks

**Comment by g1nt0ki** [comment]
@theiceeman sorry for the delay, reviewing it now

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@ node_modules
 .vscode
 .DS_store
 .env
-.idea
\ No newline at end of file
+.idea
+yarn.lock
```
Comment (g1nt0ki) [inline .gitignore]: please remove this

**Comment by llamatester** [comment]
The assetchain.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees assetchain.ts
    
    🦙 Running ASSETCHAIN.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Wed, 23 Oct 2024 05:05:30 GMT
    End Date:	Thu, 24 Oct 2024 05:05:30 GMT
    ---------------------------------------------------
    
    ASSETCHAIN 👇
    Backfill start time: 29/8/2020
    NO METHODOLOGY SPECIFIED
    Daily fees: 0
    Daily revenue: 0
    End timestamp: 1729746329 (2024-10-24T05:05:29.000Z)

File: fees/assetchain.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  const epochsToFetch = 6;
+  const epochs: any = [];
+
+  // Create array of epoch numbers to fetch (current and previous 5)
+  for (let i = 0; i < epochsToFetch; i++) epochs.push(+currentEpoch - i);
+  const epochData = await api.multiCall({ abi: snapshotABI, calls: epochs, target: contractAddress })
+  return epochData.reduce((acc: any, data: any) => acc + +data.epochFee, 0)
+}
+
+export async function getPrice() {
```
Comment (g1nt0ki) [inline fees/assetchain.ts]: the PR is mostly good except for this PR, I cant find the token on coingecko, is it traded in some exchange atm?

Comment (theiceeman) [reply fees/assetchain.ts]: Yes, it is.
There is a coinmarketcap page here for the token.
https://coinmarketcap.com/currencies/xend-finance/

Comment (theiceeman) [reply fees/assetchain.ts]: Hello @g1nt0ki
does this dispell your concerns?
can we proceed with merging the pr pls
Thanks

**Comment by dtmkeng** [comment]
@theiceeman  i try refill data but it still 0 every day  is it correct ?

**Comment by theiceeman** [comment]
Let me confirm this pls
Thank you

**Comment by theiceeman** [comment]
Hello @dtmkeng 
I reviewed this, and there was an issue with the value.
My update still returns zero, but thats because the fees generated currently are very small.
Less than 0.

It seems DefiLlama rounds values like that to 0.
Pls merge the update i made below.
Thanks.

https://github.com/DefiLlama/dimension-adapters/pull/2061

**Pull request merged**